VideoLocker/src/main/java/org/edx/mobile/base/MainApplication.java

```diff
@@ -173,14 +167,17 @@ private void tryToUpdateParseForAppUpgrade(Context context){
         if (  previousVersion < curVersion ){
             ParseInstallation installation = ParseInstallation.getCurrentInstallation();
 
-            String language = Locale.getDefault().getLanguage();
-            installation.put("preferredLanguage",language);
+            Locale locale = Locale.getDefault();
+            installation.put("preferredLanguage",locale.getLanguage());
+            installation.put("preferredCountry",locale.getCountry());
```

Comment:
@aleffert  yeah. good catch.  I am not sure if we need to check this each time application is created , or app is brought to the front?  we can register a listener to preference change. but if app is killed , it wont work. 

for now,  let's do two things together  1) when app is created, check preference,  2) add listener for change.